### ISSUE TITLE
NEXT: Tabs

### ISSUE THREAD

**Issue by endigo9740** [skeletonlabs/skeleton#2395]
This will act as a hub to centralize this information.

## Structure

React:

```html
<Tabs>
    <Tabs.List>
        <Tabs.Control name="foo">...</Tabs.Control>
        <Tabs.Control name="bar">...</Tabs.Control>
        <Tabs.Anchor>...</Tabs.Anchor>
    </Tabs.list>
    <Tabs.Content>
        <Tabs.Panel value="foo">...</Tabs.Panel>
        <Tabs.Panel value="bar">...</Tabs.Panel>
    </Tabs.Content>
</Tabs>
```

Svelte:

```html
<Tabs>
    {#snippet list()}
        <Tabs.Control name="foo">...</Tabs.Control>
        <Tabs.Control name="bar">...</Tabs.Control>
        <Tabs.Anchor>...</Tabs.Anchor>
    {/snippet}
    {#snippet content()}
        <Tabs.Panel value="foo">...</Tabs.Panel>
        <Tabs.Panel value="bar">...</Tabs.Panel>
    {/snippet}
</Tabs>
```

## Maintainer Requests

The following requests are coming straight from the Skeleton team. These are highly likely be implemented:

- [x] Should follow the [ARIA APG guidelines](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/)
- [x] Tabs should embed a radio input to control state. See v2 and our new React reference repo for help with this.
- [x] Reviewing other frameworks ([ref 1](https://ui.shadcn.com/docs/components/tabs), [ref 2](https://mantine.dev/core/tabs/)), they tend to have more explicit sectioning between the parent, list, panels, controls, and anchors. See the proposed structure in the code blocks above.
- [x] The default styling can be based on [Material Design](https://m3.material.io/components/tabs/guidelines) (wide underline) until we can provide a proper Figma mock. This should still be customizable to represent a more "enclosed" style like [Shadcn shows here](https://ui.shadcn.com/docs/components/tabs) (or our v2 "bottom nav" reference).
- [x] Tabs should support either a alignment (ex: left) or evenly spaced across the width ([ref](https://mantine.dev/core/tabs/#tabs-position))
- [x] Should support setting a default tab - which should be possible through the radio input state
- [x] should replicate most/all relevant selection event handlers from v2
- [ ] Make considerations for how icons can be implemented. Both in a above/below and side-by-side approach.
- [ ] We should support Tabs for anchor links in addition to local state management (see v2 for reference)
- [ ] Bohdan has some requests regarding more unique animations, with a sliding underline bar. We should consider this as part of the larger discussion around v3 animations that work cross-framework.

### Optional Requests

- [ ] (optional) Consider support for Tab Anchors to support [React Router](https://mantine.dev/core/tabs/#usage-with-react-router) and the [Next.js Router](https://nextjs.org/docs/pages/building-your-application/routing/redirecting). This may be redundant with the above event handlers request.
- [ ] (optional) Let's follow up on whether the React tabs can replace a couple doc-site only components we're using - including Preview and Framework selection. There's some known gotchas here.
- [ ] (optional) I'd love for the content panels to slide left/right on selection (ala Material Design). But don't stress if this is too difficult for the initial draft.

## Community Requests

The following requests have come from the community and are under consideration:
 
* n/a

## Bugs and Issues

* n/a

## Feedback

If you have additional updates or requests for this feature, please do so in the comments section below.

**Comment by endigo9740** [comment]
The Svelte component is now complete, per: https://github.com/skeletonlabs/skeleton/pull/2541#issuecomment-2027789359

The React component will come as a follow up PR next.

**Comment by endigo9740** [comment]
@Mahmoud-zino I've updated the list of requirements above to reflect the current status, as well as introduce a couple items we missed or that have come up recently. Let me know if you have questions regarding these.

**Comment by Mahmoud-zino** [comment]
@endigo9740 Follow-up on the tasks:

> Make considerations for how icons can be implemented. Both in a above/below and side-by-side approach.

This is already very flexible because users have full control of the content of tabs, so no action is needed.

> We should support Tabs for anchor links in addition to local state management (see v2 for reference)

Are you sure we want to go this route? it seems like the same problem with the avatar anchor, no control for users and the only benefit is semantic-HTML (users can always navigate to other pages using the events)

> Bohdan has some requests regarding more unique animations, with a sliding underline bar. We should consider this as part of the larger discussion around v3 animations that work cross-framework.

to achieve these transitions we have some sacrifices to make:
1. the line will be controlled by the parent component and not the children.
2. we will have to use js to get the size and offset of both the lastActiveTab and the activeTab
3. The parent component will have to save the list of tabs (references to them) so we can get the size and offset
While this seems like a very fun task, it will compromise our simplicity and no js when not needed promises.

>  (optional) I'd love for the content panels to slide left/right on selection (ala Material Design). But don't stress if this is too difficult for the initial draft.

While this doesn't require js for the transitions, we will still have to save a list of the tabs in the parent so we can decide if the transition should move left or right.

> (optional) Consider support for Tab Anchors to support [React Router](https://mantine.dev/core/tabs/#usage-with-react-router) and the [Next.js Router](https://nextjs.org/docs/pages/building-your-application/routing/redirecting). This may be redundant with the above event handlers request.

Like you mentioned, this can be achieved using the events. I will include examples for it in the docs. 

> (optional) Let's follow up on whether the React tabs can replace a couple doc-site only components we're using - including Preview and Framework selection. There's some known gotchas here.

I will create a separate PR for the docs once the react PR is merged, and then I can include the changes in the docs PR.
